docs/repo/dependencies-node-roadmap.md

```diff
@@ -1,23 +1,79 @@
+|[Transitive dependencies](#transitive-dependencies)|Dependencies brought by top level dependencies. Arbitrarily deep.|Lazily|Solution Explorer's APIs|
+
+---
 
-1. Top-level dependencies &mdash; these are the direct dependencies of the project and as such are only a single level deep.
-2. Transitive dependencies &mdash; these are dependencies that are brough in via top-level dependencies, and may be nested arbitrarily deep. For a typical project, the majority of items in its dependency graph fall into this category.
+# Top-level dependencies
 
 ## Overview
 
-### The CPS View of Dependencies
+Top level dependencies are sourced in two ways in this repo:
+
+1. We use MSBuild project evaluation and design-time build to source data about most kinds of dependencies.
+
+   |Type|Unresolved item type|Resolved item type|Comment|
+   |:-|:-|:-|:-|
+   |Analyzer|`AnalyzerReference`|`ResolvedAnalyzerReference`|Roslyn analyzers|
+   |Assembly|`Reference`|`ResolvedReference`|DLLs not sourced directly from disk|
+   |COM|`COMReference`|`ResolvedCOMReference`||
+   |Framework|`FrameworkReference`|`ResolvedFrameworkReference`|For .NET Core 2.0+ projects|
+   |Package|`PackageReference`|`ResolvedPackageReference`|NuGet|
+   |Project|`ProjectReference`|`ResolveProjectReference`||
+   |SDK|`SDKReference`|`ResolvedSDKReference`|For extension SDKs, not the .NET SDK|
+
+2. We have a dedicated pathway for shared projects, which pulls data from CPS dataflow.
+
+Additionally, third parties may provide extensions (see [Extensibily model](#extensibility-model) for more information):
+
+1. The WebTools have their own provider for NPM dependencies.
+
+## Dependencies sourced via MSBuild items
+
+This diagram gives an insight into the flow of data through the dependencies tree subsystem for top-level dependencies obtained via MSBuild.
+
+```mermaid
+%% TODO add click handlers:
+%%   click A "https://github.com/dotnet/project-system/..." "Click to view source"
+
+flowchart LR
+  subgraph UnconfiguredProject Scope
+    filters[IDependenciesSnapshotFilter]
+    handlers[IDependenciesRuleHandler]
+    subgraph ConfiguredProject Scope
+      direction TB
+      subgraph CPS Data Sources
+        evaluation([Evaluation Data])
+        design-time-build([Design-time Build Data])
+      end
+    end
+    evaluation == "snapshot + delta" ==> DependencyRulesSubscriber
+    design-time-build == "snapshot + delta" ==> DependencyRulesSubscriber
+    handlers -- "import many" -.-> DependencyRulesSubscriber
+    filters -- "import many" -.-> DependencyRulesSubscriber
+    DependencyRulesSubscriber -- "IDependenciesChanges (delta)" --> DependenciesSnapshotProvider
+    IDependenciesTreeViewProvider["IDependenciesTreeViewProvider.BuildTreeAsync"]
+    DependenciesSnapshotProvider == DependenciesSnapshot ==> IDependenciesTreeViewProvider
+    IDependenciesTreeViewProvider -- IProjectTree --> ProjectTreeProviderBase.SubmitTreeUpdateAsync
+  end
```

Comment:
This renders as:

![image](https://user-images.githubusercontent.com/350947/159216417-164d0537-5e1f-4671-8cd5-4d126a5207d7.png)